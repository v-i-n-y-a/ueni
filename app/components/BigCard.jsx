import React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import CardActions from '@material-ui/core/CardActions'
import { withStyles } from '@material-ui/core/styles'
import isEmpty from 'lodash/isEmpty'

//Using material-ui framework for responsive
const styles = {
  card: {
    margin: 20,
    display: 'flex',
    flexDirection: 'row'
  },

  location: {
    margin: '5px 0',
    fontStyle: 'italic'
  },

  description: {
    fontStyle: 'italic'
  },

  media: {
    margin: 20,
    flex: 1
  },

  content: {
    flex: 1
  },

  img: {
    width: '100%',
    minWidth: 250
  },

  link: {
    textDecoration: 'none',
    color: '#3f51b5',
    paddingLeft: '10px'
  },

  rating: {
    marginTop: 20,
    fontStyle: 'italic'
  },

  category: {
    fontStyle: 'italic'
  }
}

//#TODO: truncate description by word if it's too big add ... at the end
const BusinessCard = ({ item, classes }) => {

  if (isEmpty(item)) { return (<div></div>) }
  return (
    <Card className={classes.card}>
      <picture className={classes.media}>
        <img className={classes.img} src={item.imageUrl} />
      </picture>
      <CardContent>
        <Typography gutterBottom variant="headline" component="h2">
          {item.name}
        </Typography>
        <Typography component="p" className={classes.location}>
          {item.city}, {item.country}
        </Typography>
        <Typography component="p" className={classes.description}>
          {item.description}
        </Typography>
        <Typography component="p" className={classes.rating}>
          Rating: {item.score}
        </Typography>
        <Typography component="p" className={classes.category}>
          Category: {item.category}
        </Typography>
      </CardContent>
    </Card>
  )
}

export default withStyles(styles)(BusinessCard)
