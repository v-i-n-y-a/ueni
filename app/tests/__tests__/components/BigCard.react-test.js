import React from 'react';
import BigCard from './../../../components/BigCard.jsx';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import businessJSON from './business.json';
import { MemoryRouter } from 'react-router'

describe('<BigCard item={business} />', () => {
  it('BigCard renders correctly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <BigCard item={businessJSON} />
      </MemoryRouter>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
