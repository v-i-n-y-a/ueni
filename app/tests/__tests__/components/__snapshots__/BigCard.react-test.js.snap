// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BigCard item={business} /> BigCard renders correctly 1`] = `
<div
  className="MuiPaper-root-11 MuiPaper-elevation1-14 MuiPaper-rounded-12 MuiCard-root-10 BusinessCard-card-1"
>
  <picture
    className="BusinessCard-media-4"
  >
    <img
      className="BusinessCard-img-6"
      src="https://res.cloudinary.com/ueni/image/upload/c_scale,f_auto,fl_lossy,h_180/v1502450090/qkxvzyhladloualxcb2y.jpg"
    />
  </picture>
  <div
    className="MuiCardContent-root-38"
  >
    <h2
      className="MuiTypography-root-39 MuiTypography-headline-44 MuiTypography-gutterBottom-56"
    >
      Dani's Barber Shop
    </h2>
    <p
      className="MuiTypography-root-39 MuiTypography-body1-48 BusinessCard-location-2"
    >
      London
      , 
      United Kingdom
    </p>
    <p
      className="MuiTypography-root-39 MuiTypography-body1-48 BusinessCard-description-3"
    >
      We only use top quality products, recommended by grooming specialists and our clients themselves. Our stylish surrounding and décor shouldn’t distract you from our friendly atmosphere, we want you to feel right at home! 
    </p>
    <p
      className="MuiTypography-root-39 MuiTypography-body1-48 BusinessCard-rating-8"
    >
      Rating: 
      3.4
    </p>
    <p
      className="MuiTypography-root-39 MuiTypography-body1-48 BusinessCard-category-9"
    >
      Category: 
      Barber
    </p>
  </div>
</div>
`;
